File: ufro/src/main/java/com/ufro/model/Usuario.java
package com.ufro.model;

import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class Usuario {
    private String uid;
    private String name;
    private String phone;
    private String correo;
    private String rol; // admin / usuario
}

File: comuni_safe_front/lib/models/usuario.dart
class Usuario {
  final String uid;
  final String name;
  final String correo;
  final String phone;
  final String rol;

  Usuario({
    required this.uid,
    required this.name,
    required this.correo,
    required this.phone,
    required this.rol,
  });

  factory Usuario.fromJson(Map<String, dynamic> json) {
    return Usuario(
      uid: json['uid'] ?? '',
      name: json['name'] ?? '',
      correo: json['correo'] ?? '',
      phone: json['phone'] ?? '',
      rol: json['rol'] ?? '',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'uid': uid,
      'name': name,
      'correo': correo,
      'phone': phone,
      'rol': rol,
    };
  }
}
